src/mam/ejabberd_gen_mam_archive.erl

```diff
@@ -7,7 +7,8 @@
 -callback archive_message(_Result, jid:server(),
                           MessID :: mod_mam:message_id(), ArchiveID :: mod_mam:archive_id(),
                           LocJID :: jid:jid(), RemJID :: jid:jid(),
-                          SrcJID :: jid:jid(), Dir :: atom(), Packet :: any()) ->
+                          SrcJID :: jid:jid(), OriginID :: binary(),
+                          Dir :: atom(), Packet :: any()) ->
```

Comment:
Hmm... I remember many discussion with Radek about functions that start taking too many parameters... At some point they become unruly and hard to follow. I'd like to raise here the option to pass a map with the necessary values, it's more readable and extensible, what do you think? 🤔

Reply:
It's on my list of top thing to do after message retraction is finished. Already discussed with @michalwski and @erszcz.